### ISSUE TITLE
Background sync related crash

### ISSUE THREAD

**Issue by davidoskky** [aminecmi/ReaderforSelfoss#328]
### Description

I found another bug that causes the application to crash while caching images in background.
This is the feed https://uxmovement.com/feed

The crash seems to occur when Glide tries to load an image that isn't hosted on the website anymore.

```ruby
2021-01-10 15:16:14.161 30762-30762/apps.amine.bou.readerforselfoss W/Glide: Load failed for https://uxmovement.com/wp-content/uploads/2020/12/loginux-discloseemail.png with size [-2147483648x-2147483648]
    class com.bumptech.glide.load.engine.GlideException: Failed to load resource
      Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, REMOTE
        Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetch failed
          Cause (1 of 1): class com.bumptech.glide.load.HttpException: Not Found
2021-01-10 15:16:14.161 30762-30762/apps.amine.bou.readerforselfoss I/Glide: Root cause (1 of 1)
    com.bumptech.glide.load.HttpException: Not Found
        at com.bumptech.glide.load.data.HttpUrlFetcher.loadDataWithRedirects(HttpUrlFetcher.java:119)
        at com.bumptech.glide.load.data.HttpUrlFetcher.loadData(HttpUrlFetcher.java:54)
        at com.bumptech.glide.load.model.MultiModelLoader$MultiFetcher.loadData(MultiModelLoader.java:96)
        at com.bumptech.glide.load.model.MultiModelLoader$MultiFetcher.startNextOrFail(MultiModelLoader.java:147)
        at com.bumptech.glide.load.model.MultiModelLoader$MultiFetcher.onLoadFailed(MultiModelLoader.java:141)
        at com.bumptech.glide.load.data.HttpUrlFetcher.loadData(HttpUrlFetcher.java:60)
        at com.bumptech.glide.load.model.MultiModelLoader$MultiFetcher.loadData(MultiModelLoader.java:96)
        at com.bumptech.glide.load.engine.SourceGenerator.startNext(SourceGenerator.java:61)
        at com.bumptech.glide.load.engine.DecodeJob.runGenerators(DecodeJob.java:286)
        at com.bumptech.glide.load.engine.DecodeJob.runWrapped(DecodeJob.java:256)
        at com.bumptech.glide.load.engine.DecodeJob.run(DecodeJob.java:226)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
        at com.bumptech.glide.load.engine.executor.GlideExecutor$DefaultThreadFactory$1.run(GlideExecutor.java:347)
2021-01-10 15:16:14.162 30762-31673/apps.amine.bou.readerforselfoss E/AndroidRuntime: FATAL EXCEPTION: Thread-169
    Process: apps.amine.bou.readerforselfoss, PID: 30762
    java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Load failed
        at com.bumptech.glide.request.RequestFutureTarget.doGet(RequestFutureTarget.java:206)
        at com.bumptech.glide.request.RequestFutureTarget.get(RequestFutureTarget.java:108)
        at apps.amine.bou.readerforselfoss.api.selfoss.Item.preloadImages(SelfossModels.kt:167)
        at apps.amine.bou.readerforselfoss.background.LoadingWorker$doWork$2$onResponse$1.invoke(background.kt:132)
        at apps.amine.bou.readerforselfoss.background.LoadingWorker$doWork$2$onResponse$1.invoke(background.kt:116)
        at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
     Caused by: java.lang.IllegalStateException: Load failed
        at com.bumptech.glide.request.RequestFutureTarget.doGet(RequestFutureTarget.java:206) 
        at com.bumptech.glide.request.RequestFutureTarget.get(RequestFutureTarget.java:108) 
        at apps.amine.bou.readerforselfoss.api.selfoss.Item.preloadImages(SelfossModels.kt:167) 
        at apps.amine.bou.readerforselfoss.background.LoadingWorker$doWork$2$onResponse$1.invoke(background.kt:132) 
        at apps.amine.bou.readerforselfoss.background.LoadingWorker$doWork$2$onResponse$1.invoke(background.kt:116) 
        at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30) 
```

**Comment by aminecmi** [comment]
Thanks ! I think I found a solution.

**Comment by aminecmi** [comment]
```
val image = Glide.with(context).asBitmap()
                            .apply(glideOptions)
                            .load(url)
                            .listener(object: RequestListener<Bitmap> {
                                override fun onLoadFailed(p0: GlideException?, p1: Any?, p2: Target<Bitmap>?, p3: Boolean): Boolean {
                                    return false
                                }

                                override fun onResourceReady(p0: Bitmap?, p1: Any?, p2: Target<Bitmap>?, p3: DataSource?, p4: Boolean): Boolean {
                                    return false
                                }
                            })
                            .submit()
                            .get()
 ```
 I tried this, but it does not work.
 
 The issue here is the `.get()`, but I don't know how to handle this.

@davidoskky do you have an idea ?

**Comment by aminecmi** [comment]
Fixed in #329
